pkg/linter/linter.go

```diff
@@ -248,126 +230,126 @@ func srvLinter(_ LinterContext, path string, _ fs.DirEntry) error {
-	if isIgnoredPath(path) {
-		return nil
-	}
-
-	if !d.Type().IsRegular() {
-		// Don't worry about non-files
-		return nil
-	}
+func strippedLinter(_ string, fsys fs.FS) error {
+	return fs.WalkDir(fsys, ".", func(path string, d fs.DirEntry, err error) error {
+		if isIgnoredPath(path) {
+			return nil
+		}
 
-	info, err := d.Info()
-	if err != nil {
-		return err
-	}
+		if !d.Type().IsRegular() {
+			// Don't worry about non-files
+			return nil
+		}
 
-	if info.Size() < int64(len(elfMagic)) {
-		// This is definitely not an ELF file.
-		return nil
-	}
+		info, err := d.Info()
+		if err != nil {
+			return err
+		}
 
-	ext := filepath.Ext(path)
-	mode := info.Mode()
-	if mode&0111 == 0 && !isObjectFileRegex.MatchString(ext) {
-		// Not an executable or library
-		return nil
-	}
+		if info.Size() < int64(len(elfMagic)) {
+			// This is definitely not an ELF file.
+			return nil
+		}
 
-	f, err := lctx.fsys.Open(path)
-	if err != nil {
-		return fmt.Errorf("opening file: %w", err)
-	}
-	defer f.Close()
+		ext := filepath.Ext(path)
+		mode := info.Mode()
+		if mode&0111 == 0 && !isObjectFileRegex.MatchString(ext) {
+			// Not an executable or library
+			return nil
+		}
 
-	// Both os.DirFS and go-apk return a file that implements ReaderAt.
-	// We don't have any other callers, so this should never fail.
-	readerAt, ok := f.(io.ReaderAt)
-	if !ok {
-		return fmt.Errorf("fs.File does not impl ReaderAt: %T", f)
-	}
+		f, err := fsys.Open(path)
+		if err != nil {
+			return fmt.Errorf("opening file: %w", err)
+		}
+		defer f.Close()
 
-	hdr := make([]byte, len(elfMagic))
-	if _, err := readerAt.ReadAt(hdr, 0); err != nil {
-		return fmt.Errorf("failed to read %d bytes for magic ELF header: %w", len(elfMagic), err)
-	}
+		// Both os.DirFS and go-apk return a file that implements ReaderAt.
+		// We don't have any other callers, so this should never fail.
+		readerAt, ok := f.(io.ReaderAt)
+		if !ok {
+			return fmt.Errorf("fs.File does not impl ReaderAt: %T", f)
+		}
 
-	if !bytes.Equal(elfMagic, hdr) {
-		// No magic header, definitely not ELF.
-		return nil
-	}
+		hdr := make([]byte, len(elfMagic))
+		if _, err := readerAt.ReadAt(hdr, 0); err != nil {
+			return fmt.Errorf("failed to read %d bytes for magic ELF header: %w", len(elfMagic), err)
+		}
 
-	file, err := elf.NewFile(readerAt)
-	if err != nil {
-		// We don't particularly care if this fails otherwise.
-		fmt.Printf("WARNING: Could not open file %q as executable: %v\n", path, err)
-	}
-	defer file.Close()
+		if !bytes.Equal(elfMagic, hdr) {
+			// No magic header, definitely not ELF.
+			return nil
+		}
 
-	// No debug sections allowed
-	if file.Section(".debug") != nil || file.Section(".zdebug") != nil {
-		return fmt.Errorf("ELF file is not stripped")
-	}
+		file, err := elf.NewFile(readerAt)
+		if err != nil {
+			// We don't particularly care if this fails otherwise.
+			fmt.Printf("WARNING: Could not open file %q as executable: %v\n", path, err)
```

Comment:
If this is easy enough to disable, I'd rather return an error here than pollute stdout. I think this is a holdover from when we didn't have the check right above that skips non-elf-magic.